packages/teleterm/src/ui/services/connectionTracker/connectionTrackerService.ts

```diff
@@ -129,29 +129,32 @@ export class ConnectionTrackerService extends ImmutableStore<ConnectionTrackerSt
       .disconnect();
   }
 
-  removeItem(id: string): void {
+  async removeItem(id: string): Promise<void> {
     const connection = this.state.connections.find(c => c.id === id);
     if (!connection) {
       return;
     }
 
-    this._trackedConnectionOperationsFactory.create(connection).remove?.();
+    await this._trackedConnectionOperationsFactory
+      .create(connection)
+      .remove?.();
```

Comment:
Adding a noop `remove` to non-k8s connections would be a valid approach here too IMHO.

Reply:
Done.